commit 4183fbbb8f9d8fbc1266a77867b586bfea0c0e5c (refs/changes/48/1448/2)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2017-02-07T18:02:11+00:00 (2 years, 8 months ago)
    
    joyent/node-mname#13 large amount of time spent in modulo operations
    Reviewed by: Cody Peter Mello <cody.mello@joyent.com>

diff --git a/lib/dns-buffer.js b/lib/dns-buffer.js
index 5d3a3bc..08ab3ac 100644
--- a/lib/dns-buffer.js
+++ b/lib/dns-buffer.js
@@ -191,21 +191,21 @@ DNSBuffer.prototype.writeNamePlain = function (name) {
 DNSBuffer.prototype.writeUInt32 = function (v) {
         while (this._offset + 4 > this._size)
                 this.expand();
-        this._buffer.writeUInt32BE(v % 0xffffffff, this._offset);
+        this._buffer.writeUInt32BE(v, this._offset);
         this._offset += 4;
 };
 
 DNSBuffer.prototype.writeUInt16 = function (v) {
         while (this._offset + 2 > this._size)
                 this.expand();
-        this._buffer.writeUInt16BE(v & 0xffff, this._offset);
+        this._buffer.writeUInt16BE(v, this._offset);
         this._offset += 2;
 };
 
 DNSBuffer.prototype.writeUInt8 = function (v) {
         while (this._offset + 1 > this._size)
                 this.expand();
-        this._buffer[this._offset++] = v & 0xff;
+        this._buffer.writeUInt8(v, this._offset++);
 };
 
 DNSBuffer.prototype.write = function (buf) {
